test/libsolutil/JSON.cpp

```diff
@@ -33,6 +33,48 @@ namespace solidity::util::test
 
 BOOST_AUTO_TEST_SUITE(JsonTest, *boost::unit_test::label("nooptions"))
 
+BOOST_AUTO_TEST_CASE(json_types)
+{
+	auto check = [](Json::Value value, string const& expectation) {
+		BOOST_CHECK(jsonCompactPrint(value) == expectation);
+	};
+
+	Json::Value value;
+	BOOST_CHECK(value.empty());
+	value = {};
+	BOOST_CHECK(value.empty());
+	value = Json::Value();
+	BOOST_CHECK(value.empty());
+	value = Json::nullValue;
+	BOOST_CHECK(value.empty());
+
+	check(value, "null");
+	check({}, "null");
+	check(Json::Value(), "null");
+	check(Json::nullValue, "null");
+	check(Json::objectValue, "{}");
+	check(Json::arrayValue, "[]");
+	check(Json::UInt(1), "1");
+	check(Json::UInt(-1), "4294967295");
+	check(Json::UInt64(1), "1");
+	check(Json::UInt64(-1), "18446744073709551615");
```

Comment:
Yes, we use this, and this is actually output which can't be properly parsed by JavaScript frameworks. It is used for gas estimation for example.

Reply:
But that basically means that the range of integers supported in compliant JSON is less than we need it to support, isn't it? Do we want to switch to printing ints out of the supported range as strings or do we prefer to stay non-compliant in this regard?

Reply:
[ECMA-404](https://www.ecma-international.org/wp-content/uploads/ECMA-404_2nd_edition_december_2017.pdf) doesn't care about numbers:

>JSON is agnostic about the semantics of numbers. In any programming language, there can be a variety of number types of various capacities and complements, fixed or floating, binary or decimal. That can make interchange between different programming languages difficult. JSON instead offers only the representation of numbers that humans use: a sequence of digits. All programming languages know how to make sense of digit sequences even if they disagree on internal representations. That is enough to allow interchange.

However if we want to import JSON in Javascript, then it gets iffy if numbers are larger than `2^53-1`.